resources/lib/modules/addon.py

```diff
@@ -169,37 +167,36 @@ def _prepare_for_data():
         sock.listen(1)
         return sock
 
-    @staticmethod
-    def _wait_for_data(sock, timeout=10):
+    def _wait_for_data(self, sock, timeout=10):
         """ Wait for data to arrive on the socket """
         # Set a connection timeout
         # The remote and should connect back as soon as possible so we know that the request is being processed
         sock.settimeout(timeout)
 
         # Accept one client
         try:
-            _LOGGER.debug('Waiting for a connection on port %s...', sock.getsockname()[1])
+            _LOGGER.debug('Waiting for a connection from %s on port %s...', self.addon_id, sock.getsockname()[1])
             conn, addr = sock.accept()
             _LOGGER.debug('Connected to %s:%s! Waiting for result...', addr[0], addr[1])
 
             # Read until the remote end closes the connection
-            buffer = ''
+            buf = ''
             while True:
                 chunk = conn.recv(1024)
                 if not chunk:
                     break
-                buffer += chunk
+                buf += chunk
 
-            if not buffer:
+            if not buf:
                 # We got an empty reply, this means that something didn't go according to plan
-                raise Exception('Something went wrong in the other Add-on')
+                raise Exception('Something went wrong in %s' % self.addon_id)
 
-            return buffer
+            return buf
 
         except socket.timeout:
-            raise Exception('Timout waiting on reply from other Add-on')
+            raise Exception('Timout waiting for reply from %s on port %s' % (self.addon_id, sock.getsockname()[1]))
 
         finally:
             # Close our socket
-            _LOGGER.debug('Closing socket on port %s', sock.getsockname()[1])
+            _LOGGER.debug('Closing socket on port %s' % sock.getsockname()[1])
```

Comment:
Dit moet niet. Bij logging moet je , gebruiken ipv %.

Op die manier moet die de string interpolation ook pas doen als er effectief gelogd wordt.

Reply:
Maar normaal geeft is er een pylint test die daar over klaagt.

Reply:
Ja, dat was niet de bedoeling. Ik was naar pyformat overgeschakeld eerst omdat e substitutie eerst wat ingewikkelder was, maar dan beslist om het eenvoudiger te houden. En de Exception() hadden wel een % nodig, de logger instances niet. Dus ik heb niet gezien dat ik deze verkeerd had gedaan.